core/domain/cron_services.py

```diff
@@ -56,6 +63,40 @@ def get_stuck_jobs(recency_msecs):
     return stuck_jobs
 
 
+def delete_models_marked_as_deleted():
+    """Hard-delete all models that are marked as deleted (have deleted field set
+    to True) and were last updated more than eight weeks ago.
+    """
+    date_now = datetime.datetime.utcnow()
+    date_before_which_hard_delete = (
+        date_now - PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED)
+    for model_class in models.Registry.get_all_storage_model_classes():
+        deleted_models = model_class.query(
+            model_class.deleted == True  # pylint: disable=singleton-comparison
+        ).fetch()
+        models_to_hard_delete = [
+            deleted_model for deleted_model in deleted_models
+            if deleted_model.last_updated < date_before_which_hard_delete
+        ]
+        model_class.delete_multi(models_to_hard_delete)
+
+
+def mark_models_as_deleted():
+    """Mark some types of models (that we shouldn't keep for long time)
+    as deleted if they were last updated more than four weeks ago.
+    """
+    date_before_which_mark = (
+        datetime.datetime.utcnow() - PERIOD_TO_MARK_MODELS_AS_DELETED)
+    for model_class in MODEL_CLASSES_TO_MARK_AS_DELETED:
+        models_to_mark_as_deleted = model_class.query(
+            model_class.last_updated < date_before_which_mark
+        ).fetch()
+        for model_to_mark_as_deleted in models_to_mark_as_deleted:
```

Comment:
Note (optional): OK to do `for model in models_to_mark_as_deleted`

Reply:
Done.

Reply:
Not done. It's optional, but please don't say "Done" when it's not done -- that's not right.

Reply:
Yep, sorry.